File: backend/module/Radio/src/Radio/Entity/Author.php
<?php

namespace Radio\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity 
 * @ORM\Table(name="author")
 * */
class Author {

    /**
     * @ORM\Id 
     * @ORM\Column(type="integer") 
     * @ORM\GeneratedValue 
     * */
    protected $id;

    /**
     * @ORM\Column(type="string") 
     * */
    protected $name;

    /**
     * @ORM\Column(type="string", nullable=true)
     * */
    protected $email;
    
    /**
     * @ORM\Column(type="string", length=60, nullable=true)
     * */
    protected $alias;
    
    /**
     * @ORM\Column(type="string",length=50)
     * */
    protected $photo;
    
    /**
     * @ORM\Column(type="string",length=20) 
     * */
    protected $avatar;
    
    /**
     * @ORM\Column(type="text") 
     * */
    protected $introduction = "";

    /**
     * @ORM\OneToMany(targetEntity="Contribution",mappedBy="author", fetch="EAGER")
     */
    protected $contributions;
    
    /**
     * @ORM\OneToOne(targetEntity="User")
     * @ORM\JoinColumn(name="user_id", referencedColumnName="id")
     */
    protected $user;
    
    /**
     * @ORM\ManyToMany(targetEntity="Url")
     * @ORM\JoinTable(name="author_url",
     *      joinColumns={@ORM\JoinColumn(name="author_id", referencedColumnName="id")},
     *      inverseJoinColumns={@ORM\JoinColumn(name="url_id", referencedColumnName="id", unique=true)}
     *      )
     **/
    protected $urls = array();

    public function getId() {
        return $this->id;
    }

    public function setId($id) {
        $this->id = $id;
    }

    public function getName() {
        return $this->name;
    }

    public function setName($name) {
        $this->name = $name;
    }

    public function getPhoto() {
        return $this->photo;
    }
    
    public function setPhoto($photo) {
        $this->photo = $photo;
    }
    
    public function getAvatar() {
        return $this->avatar;
    }
    
    public function setAvatar($avatar) {
        $this->avatar = $avatar;
    }
    
    public function getIntroduction() {
        return $this->introduction;
    }
    
    public function setIntroduction($introduction) {
        $this->introduction = $introduction;
    }
    
    public function getUser() {
        return $this->user;
    }
    
    public function setUser($user) {
        $this->user = $user;
    } 
    
  
    public function getContributions() {
        return $this->contributions;
    }

    public function setContributions($contributions) {
        $this->contributions = $contributions;
    }

    public function getAlias() {
        return $this->alias;
    }

    public function setAlias($alias) {
        $this->alias = $alias;
    }

    
    public function toArray() {
        $a = $this->toArrayShort();
        $a['introduction'] = $this->getIntroduction();
        $a['photo'] = $this->getPhoto();
        return $a;
    }

    public function toArrayShort() {
        $a = array();
        $a['id'] = $this->getId();
        $a['name'] = $this->getName();
        $a['avatar'] = $this->getAvatar();
        $a['alias'] = $this->getAlias();
        return $a;
    }
    
    public function getUrls() {
        return $this->urls;
    }
    
    public function addUrl($url) {
        $this->urls[] = $url;
    }
    
    public function setUrls($urls) {
        $this->urls = $urls;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email)
    {
        $this->email = $email;
    }

    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }


}

?>

File: util/model/src/main/java/hu/radio/tilos/model/Author.java
package hu.radio.tilos.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity()
@Table(name = "author")
public class Author {

    @Id
    private int id;

    @Basic
    @Column
    private String name;

    @Basic
    @Column
    private String alias;

    @Basic
    @Column
    private String photo;

    @Basic
    @Column
    private String avatar;

    @Basic
    @Column
    private String introduction;

    @Basic
    @Column
    private String email;

    @OneToOne
    private User user;

    @OneToMany(mappedBy = "author")
    private List<Contribution> contributions = new ArrayList<Contribution>();

    public List<Contribution> getContributions() {
        return contributions;
    }

    public void setContributions(List<Contribution> contributions) {
        this.contributions = contributions;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getIntroduction() {
        return introduction;
    }

    public void setIntroduction(String introduction) {
        this.introduction = introduction;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}
